### PR TITLE
MSVC: Update to latest buildscript version.

### PR DESCRIPTION
**Pull request by Ghabry**
Add xmplite and wildmidi. Fix ARM build.

Not ready yet. My final build test isn't finished :)

Checklist what I have tested by now:

x86 Debug
 - [x] toolchain
 - [x] liblcf
 - [x] Player
 - [x] WildMidi works
 - [x] xmplite works

x86 Release
 - [x] toolchain
 - [x] liblcf
 - [x] Player
 - [x] WildMidi works
 - [x] xmplite works

amd64 Debug
 - [x] toolchain
 - [x] liblcf
 - [x] Player
 - [x] WildMidi works
 - [x] xmplite works

amd64 Release
 - [x] toolchain
 - [x] liblcf
 - [x] Player
 - [x] WildMidi works
 - [x] xmplite works

ARM Debug
 - [x] toolchain
 - [x] liblcf
 - [x] Player

ARM Release
 - [x] toolchain
 - [x] liblcf
 - [x] Player

### CODE REVIEW

File: builds/vs2015/PlayerLib.vcxproj [hunk 1]
```diff
@@ -752,4 +756,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
+</Project>
```
Comment (carstene1ns) [inline builds/vs2015/PlayerLib.vcxproj]: Did you use MSVC to save this file? If not, I would like to keep the EOF inebreak.

File: builds/vs2015/PlayerLib.vcxproj.filters [hunk 2]
```diff
@@ -1006,5 +1012,11 @@
...
     </ClInclude>
+    <ClInclude Include="..\..\src\decoder_wildmidi.h">
+      <Filter>Source Files\Backend\Audio</Filter>
+    </ClInclude>
+    <ClInclude Include="..\..\src\decoder_xmp.h">
+      <Filter>Source Files\Backend\Audio</Filter>
+    </ClInclude>
   </ItemGroup>
-</Project>
+</Project>
```
Comment (carstene1ns) [inline builds/vs2015/PlayerLib.vcxproj.filters]: same here.

File: src/registry.cpp [hunk 3]
```diff
@@ -36,6 +36,7 @@
 #endif
 
 std::string Registry::ReadStrValue(HKEY hkey, std::string const& key, std::string const& val, REGVIEW view) {
+#ifndef _ARM_
```
Comment (carstene1ns) [inline src/registry.cpp]: Do you intend to add registry reading on WINRT in the future?
If not, instead of patching the functions here, the filefinder could be changed and this whole file excluded on ARM.

File: builds/vs2015/PlayerLib.vcxproj (hunk 1 above)
Comment (Ghabry) [inline builds/vs2015/PlayerLib.vcxproj]: Yeah this is MSVC behaviour. Can manually insert a line break again...

File: src/registry.cpp (hunk 3 above)
Comment (Ghabry) [inline src/registry.cpp]: The ARM toolchain doesn't export symbols for this. Will improve the stubbing to match other port behaviour.

Comment (Ghabry) [inline src/registry.cpp]: Besides that it's a bit hard to get RTP entries in the registry without a ARM installer :D

File: builds/vs2015/PlayerLib.vcxproj (hunk 1 above)
Comment (carstene1ns) [inline builds/vs2015/PlayerLib.vcxproj]: Is okay then, I just thought it was MSVC that added it in #1100. I am just against unnecessary whitespace changes and this case is likely to get changed over and over again.

File: src/registry.cpp (hunk 3 above)
Comment (carstene1ns) [inline src/registry.cpp]: Well, registry values could be added and files copied over manually 😀

**Comment by Ghabry** [comment]
WildMidi doesn't work because the API passes an Ansi string to the Unicode API :/

Additionally they use FindFirstFile to get size information about a file but for this you could also use GetFileAttributes o.O
And afterwards they use the non-unicode aware "open" function :). WINAPI is hard ;)... going to submit a patch later.

**Comment by Ghabry** [comment]
Sorry for the noise, shouldn't do boolean algebra at this time of the day *yawn*

**Comment by Ghabry** [comment]
closing Player while xmp plays something throws an assertion: "Ausnahmefehler bei 0x00007FF8A07E2208 (ucrtbase.dll) in Player64.exe: Ein ungültiger Parameter wurde an eine Funktion übergeben, die ungültige Parameter als schwerwiegend einstuft." (invalid arg was given to a function that considers invalid args as serious)

In
```
 	ucrtbase.dll!_invoke_watson()	Unbekannt
 	ucrtbase.dll!_invalid_parameter()	Unbekannt
 	ucrtbase.dll!_invalid_parameter_noinfo()	Unbekannt
 	ucrtbase.dll!__longjmp_internal()	Unbekannt
 	ucrtbase.dll!_fclose_nolock()	Unbekannt
 	ucrtbase.dll!fclose()	Unbekannt
 	ucrtbase.dll!_fcloseall()	Unbekannt
 	ucrtbase.dll!__acrt_uninitialize_stdio()	Unbekannt
 	ucrtbase.dll!__acrt_initialize_fma3()	Unbekannt
 	ucrtbase.dll!__acrt_execute_uninitializers()	Unbekannt
 	ucrtbase.dll!iswctype()	Unbekannt
 	ntdll.dll!LdrpCallInitRoutine()	Unbekannt
 	ntdll.dll!LdrShutdownProcess()	Unbekannt
 	ntdll.dll!RtlExitUserProcess()	Unbekannt
 	kernel32.dll!ExitProcessImplementation()	Unbekannt
 	ucrtbase.dll!exit()	Unbekannt
 	ucrtbase.dll!exit()	Unbekannt
>	Player64.exe!__scrt_common_main_seh() Zeile 262	C++
 	kernel32.dll!BaseThreadInitThunk()	Unbekannt
 	ntdll.dll!RtlUserThreadStart()	Unbekannt
```

**Comment by Ghabry** [comment]
My buildscript PR is ready. Can confirm that everything works :)

**Comment by carstene1ns** [comment]
Jenkins: test this please!
